asciidoc/courses/genai-mcp-build-custom-tools-python/modules/1-getting-started/lessons/4-mcp-inspector/lesson.adoc

```diff
@@ -0,0 +1,178 @@
+= The MCP Inspector
+:type: lesson
+:order: 4
+
+// * What is the MCP Inspector?
+// * Using MCP Inspector in Codespaces
+// * Testing tools and resources with the Inspector
+
+Great!  You now have your first MCP server running.  But the Chat window in VS Code isn't the most efficient or cost-effective way to test your server.
+
+In this lesson, you will learn about the MCP Inspector, a tool that allows you to test and debug your MCP servers and tools through a UI.
+
+== Introducing the MCP Inspector
+
+The MCP Inspector is a tool that allows you to test and debug your MCP servers and tools through a UI.
+
+// TODO: screenshot with strawberry tool
+image::images/mcp-inspector.png[The MCP Inspector UI]
+
+The left hand panel allows you to define the connection details for the MCP server you want to test.
+
+Once connected to a server, the server features are listed to the right.
+
+
+=== Running the MCP Inspector
+
+The tool is written in JavaScript, so link:https://nodejs.org/en/download[you will need Node.js and the `npx` command installed^].
+
+[source,shell]
+.Running the MCP Inspector
+----
+npx @modelcontextprotocol/inspector
+----
+
+The first time you run the tool with `npx`, you will be required to install the tool.
+
+[source]
+.Installing the MCP Inspector
+----
+Need to install the following packages:
+@modelcontextprotocol/inspector@0.17.1
+Ok to proceed? (y) 
+----
+
+Press `y` to proceed.
+
+[TIP]
+.Using `npx`
+====
+The `npx` command is a package runner for Node.js.  It is used to run Node.js packages without installing them globally.
+
+You can install npx along with Node.js by link:https://nodejs.org/en/download[downloading Node.js from the official website^].
+====
+
+Once installed, the MCP Inspector UI will be served on port 6274.
+
+[source]
+----
+Starting MCP inspector...
+⚙️ Proxy server listening on localhost:6277
+🔑 Session token: d3a6bc6d116f75b1a5f7861361fb5fa7190c29b02c70f614e8ccf14080d1eadd
+   Use this token to authenticate requests or set DANGEROUSLY_OMIT_AUTH=true to disable auth
+
+🚀 MCP Inspector is up and running at:
+   http://localhost:6274/?MCP_PROXY_AUTH_TOKEN=d3a6bc6d116f75b1a5f7861361fb5fa7190c29b02c70f614e8ccf14080d1eadd
+
+🌐 Opening browser...
+----
+
+Note the Session token, you may need to copy and paste this into the Configuration tab of the MCP Inspector to connect to your server and run tools.
+
+==== Pre-filling the Server details 
```

Comment:
I would suggest moving this further down after they have used mcp inspector. At this point the learner doesnt know that you have to fill your server details in.